hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcess.java

```diff
@@ -0,0 +1,153 @@
+package pl.allegro.tech.hermes.consumers.supervisor.process;
+
+import org.jctools.queues.SpscArrayQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.consumers.consumer.Consumer;
+
+import java.time.Clock;
+import java.util.Objects;
+
+public class ConsumerProcess implements Runnable {
+
+    private static final Logger logger = LoggerFactory.getLogger(ConsumerProcess.class);
+
+    private final SpscArrayQueue<Signal> signals = new SpscArrayQueue<>(100);
```

Comment:
In my opinion 100 capacity seems a little dangerous because we won't accept more than 100 events that subscription was assigned for instance. That could happen especially on bootstrap. I would increase increase this to 100*100 or made it configurable.